template/web/src/hooks/useCollectionMetadata.ts

```diff
@@ -1,115 +1,145 @@
-import { useCallback, useEffect, useState } from 'react';
+import { useQuery } from '@tanstack/react-query';
 import { Abi, Address } from 'abitype';
-import { useReadContract } from 'wagmi';
+import { Chain } from 'viem/chains';
+import { useAccount, usePublicClient } from 'wagmi';
 import { ipfsToHTTP } from '@/utils/ipfs';
 
+// Minimal ABI with contractURI view function
+const AbiWithContractURI = [
+  {
+    inputs: [],
+    name: 'contractURI',
+    outputs: [{ internalType: 'string', name: '', type: 'string' }],
+    stateMutability: 'view',
+    type: 'function',
+  },
+] as const;
+
+// Minimal ABI with uri view function
+const AbiWithUri = [
+  {
+    inputs: [{ internalType: 'uint256', name: 'id', type: 'uint256' }],
+    name: 'uri',
+    outputs: [{ internalType: 'string', name: '', type: 'string' }],
+    stateMutability: 'view',
+    type: 'function',
+  },
+] as const;
+
 /**
  * There is some differences in URI standards between ERC721 and 1155, handle those in this component.
  */
-enum UriFunctionType {
-  uri = 'uri',
-  contractURI = 'contractURI',
-}
-// A future enhancement would be to track error state from the contract read
-// and the fetch so that we can gracefully surface issues to users.
-type CollectionMetadataResult =
-  | {
-      collectionName: null;
-      description: null;
-      imageAddress: null;
-      isLoading: true;
-    }
-  | {
-      collectionName: string;
-      description: string;
-      imageAddress: string;
-      isLoading: false;
-    };
+const uriFunctionType = {
+  uri: 'uri',
+  contractURI: 'contractURI',
+} as const;
+
+const getIsAbiWithContractURIViewFunction = (abi: Abi) =>
+  !!abi.find((el) => el.type === 'function' && el.name === uriFunctionType.contractURI);
+const getIsAbiWithUriViewFunction = (abi: Abi) =>
+  !!abi.find((el) => el.type === 'function' && el.name === uriFunctionType.uri);
+
+const isValidAbi = (abi: Abi) =>
```

Comment:
I think lets move these functions to utils because they can be used across multiple files

Reply:
answered in another comment regarding splitting these away from the hook